modules/programs/i3status.nix

```diff
@@ -0,0 +1,206 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.i3status;
+
+  enabledModules = filterAttrs (n: v: v.enable) cfg.modules;
+
+  formatOrder = n: ''order += "${n}"'';
+
+  formatModule = n: v:
+    let
+      formatLine = n: v:
+        let
+          formatValue = v:
+            if isBool v then
+              (if v then "true" else "false")
+            else if isString v then
+              ''"${v}"''
+            else
+              toString v;
+        in "${n} = ${formatValue v}";
+    in ''
+      ${n} {
+        ${concatStringsSep "\n  " (mapAttrsToList formatLine v)}
+      }
+    '';
+
+  settingsType = with types; attrsOf (oneOf [ bool int str ]);
+
+  sortAttrNamesByPosition = comparator: set:
+    let pos = n: set."${n}".position;
+    in sort (a: b: comparator (pos a) (pos b)) (attrNames set);
+in {
+  options.programs.i3status = {
+    enable = mkEnableOption "i3status";
+
+    enableDefault = mkOption {
```

Comment:
I don't quite see the benefit of this option. Why not just use the `default` field of the `programs.i3status.general` and `programs.i3status.modules` options?

Also are the defaults supposed to match the ones used by upstream i3status when no configuration is given? If so, is it likely to remain compatible across versions?

Reply:
Defaults are set in `config` in accordance with [RFC 42](https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#default-values), regardless of `enableDefault`. The `enableDefault` option is just a convenience, so you don't have to add

```
programs.i3status.modules = {
  ipv6.enable = false;
  "wireless _first_".enable = false;
  "ethernet _first_".enable = false;
  "battery all".enable = false;
  "disk /".enable = false;
  load.enable = false;
  memory.enable = false;
  "tztime local".enable = false;
};
```

to start with a bare configuration. It isn't strictly necessary. I'm fine with removing it, if you think that would be better.

The configuration is the [default i3status configuration](https://github.com/i3/i3status/blob/master/i3status.conf). I think i3status is pretty stable, so I don't expect it to be incompatible any time soon, if ever.